test/src/test/groovy/hudson/model/AbstractProjectTest.groovy

```diff
@@ -594,4 +596,37 @@ public class AbstractProjectTest extends HudsonTestCase {
         }
         return con
     }
+
+    @Issue("JENKINS-27549")
+    public void testLoadingWithNPEOnTriggerStart() {
+        AbstractProject project = jenkins.createProjectFromXML("foo", getClass().getResourceAsStream("AbstractProjectTest/npeTrigger.xml"))
+
+        assert project.triggers().size() == 1
+    }
+
+    @TestExtension
```

Comment:
Actually `@TestExtension` should be on the descriptor only, not the `Trigger` implementation class. (The extension is the singleton descriptor. Instances of it are not extensions.)